src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs

```diff
@@ -247,11 +254,53 @@ private void FireOnCompleted()
             }
         }
 
-
         public void Write(ArraySegment<byte> data, Action<Exception, object> callback, object state)
         {
             ProduceStart(immediate: false);
-            SocketOutput.Write(data, callback, state);
+
+            if (_autoChunk)
```

Comment:
Watch out for zero-length writes - when chunking is active they can look like an end-of-stream in the middle of the response.